database.py

```diff
@@ -65,117 +89,127 @@ def insertPost(self, user_data):
+		query_filter = {'_id': id}
 		if attr:
 			attr_json = json.loads(attr);
 			for key, value in attr_json:
 				query_filter[key] = value
 
-		data = self.user_table.find_one(query_filter)
+		data = {}
+		if table == 'user':
+			data = self.user_table.find_one(query_filter)
+		else:
+			data = self.prog_table.find_one(query_filter)
 
 		return data
 
-	def deletePost(self, user_id):
+	def deletePost(self, table, id):
 		"""
 		Input parameter:
-			user_id: int
+			table: table to insert user data, should be either "program" or "user"
+			id: int
 		Output parameter: 
 			boolean indicate is delete post is successful
 
-		Remove a document from DB by user_id if user_id exist.
+		Remove a document from DB by id if id exist.
 		"""
-		if not self.checkExist(user_id):
+		if not self.checkExist(table, id) or not self.validateTable(table):
 			return False
 
-		query_filter = {'_id': user_id}
-		self.user_table.remove(query_filter)
+		query_filter = {'_id': id}
+		if table == 'user': 
+			self.user_table.remove(query_filter)
+		else:
+			self.prog_table.remove(query_filter)
 		return True
 
-	def checkExist(self, user_id):
+	def checkExist(self, table, id):
 		""" 
 		Input parameter: 
-			user_id: int
+			table: table to insert user data, should be either "program" or "user"
+			id: int
 		Output parameter: 
-			boolean indicates if the user_id info exist or not
+			boolean indicates if the id info exist or not
 
 		Check if the document exist in DB or not. Ture means exist, false means no.
 		"""
 
-		query_filter = {'_id': user_id}
-		cnt = self.user_table.find(query_filter).count()
+		query_filter = {'_id': id}
+		cnt = 0
+		if table == 'user':
+			cnt = self.user_table.find(query_filter).count()
+		else:
+			cnt = self.prog_table.find(query_filter).count()
+
 		if (cnt > 0):
 			return True
 
 		return False
 
-	def update(self, user_id, user_data):
+	def validateTable(self, table):
+		
+		return (table == 'user') or (table == 'program')
+
+	def update(self, table, id, user_data):
 		"""
 		Input parameter:
-			user_id: integer
+			table: table to insert user data, should be either "program" or "user"
+			id: integer
 			data: string, must be a kay value pair JSON
 		Output parameter:
 			boolean indicates if update is successful or not
 		"""
 
-		if not self.checkExist(user_id):
+		if not self.checkExist(table, id) or not self.validateTable(table):
 			return False
 
 		query_filter = {}
-		query_filter['_id'] = user_id
-		input_data = {'$set': json.loads(user_data)}
+		query_filter['_id'] = id
 
-		print(input_data)
-		#data = json.loads(user_data)
-		#data['_id'] = user_id
-		#print(data)
-		#for key, value in data:
-		#	input_data[key] = value
+		input_data = {'$set': user_data}
 
-		self.user_table.update(query_filter, input_data)
+		if table == 'user':
+			self.user_table.update(query_filter, input_data)
+		else:
+			self.prog_table.update(query_filter, input_data)
 
 		return True
 
 if __name__ == "__main__":
```

Comment:
Can you please put all the unittests into a different test file? Please cover all the test cases in the unittest.